### PR TITLE
Upgrade to SDK 46

### PR DESCRIPTION
**Pull request by keith-kurak**
### What
- Updated all examples to SDK 46
- Checked for removed libraries in examples per release blog (none found)
- Added a quick-and-dirty **upgrade-dependencies.sh** script that runs `expo upgrade` on all projects at once and writes the command output to a non-committed folder for review (can also run `expo doctor --fix-dependencies` on all projects, for those times where the SDK's RN version is updated mid-PR :smile: ).
- Spot-checked several projects based on a) if they failed to automatically update via the aforementioned script, b) were mentioned specifically during the previous update (https://github.com/expo/examples/pull/371), or c) otherwise seemed interesting or notable.
- In some cases, updated additional dependencies and/ or readme's to reflect my experiences getting the spot-checked examples up-and-running
- Made readme's consistently refer to `expo start` across the board due to several **package.json** files containing no `scripts`.
- Remove references to global CLI and `eject`.

#### :white_check_mark: with-yarn-workspaces
- updated to match the instructions in the [Expo Monorepos Guide](https://docs.expo.dev/guides/monorepos/)
- tweaked the npm scripts to account for removal of `expo-yarn-workspaces`, added separate script for watching packages (previously ran automatically after running `yarn`)
#### :white_check_mark: with-electron
- I've tried adding electron at least three times in recent months, and it never works unless I run `yarn expo-electron customize` and add `contextIsolation: false` to **index.js**, so updated accordingly. Wondering if this should either be reflected in docs or if `@expo/electron-adapter` can be/ should be updated to do this by default.
#### :white_check_mark: with-gatsby
- Kept `gatsby` version at 2.x, as plugin isn't compatible with 3 or 4.
#### :white_check_mark: with-three / with-react-three-fiber
- This basic example works with Expo packages that are more recent than the peer deps specify in `expo-three`, plus a post 0.127.0 version of `three`, which fixes https://github.com/expo/expo-three/issues/202
#### :warning: with-splash-screen
- Works on Android, but `Updates.reloadAsync()` fails silently with "You cannot reload when expo-updates is not enabled" on iOS on Expo Go (not a problem with example,[ issue opened](https://linear.app/expo/issue/ENG-6012/you-cannot-reload-when-expo-updates-is-not-enabled-error-on-expo-go))
#### :warning: with-nextjs
- Works if I leave `@expo/next-adapter`/ `nextjs` dependencies on the same version; homepage url hangs indefinitely if I run `yarn expo-next` from scratch on latest adapter version (which installs next 12.x, which uses requires webpack 5, while the adapter uses 4), I get a fatal babel error if I install the required peer-dependency version of next (^11, which can use Webpack 4, but doesn't by default).
#### :white_check_mark: with-drawer-navigation
- Required Babel config change due to open issue with Reanimated: https://github.com/software-mansion/react-native-reanimated/issues/3410

### CODE REVIEW

File: blank/README.md [hunk 1]
```diff
@@ -19,18 +19,17 @@
 
 - Install packages with `yarn` or `npm install`.
   - If you have native iOS code run `npx pod-install`
-- Run `yarn start` to start the bundler.
+- Run `npx expo start` to start the bundler.
```
Comment (EvanBacon) [inline blank/README.md:22]: Doesn't matter but we inject the start script when the user bootstraps with CRNA.

Comment (keith-kurak) [reply blank/README.md:22]: good to know! Some examples were one way, some the other, so was trying to pick one and apply it across the board. Knowing that it's always there, makes sense to it back to yarn/ npm, which seems more consistent with other templates.

File: blank/README.md [hunk 2]
```diff
@@ -19,18 +19,17 @@
...
 - Open the project in a React runtime to try it:
   - iOS: [Client iOS](https://itunes.apple.com/app/apple-store/id982107779)
   - Android: [Client Android](https://play.google.com/store/apps/details?id=host.exp.exponent&referrer=blankexample)
   - Web: Any web browser
 
-## Adding Native Code
+## Running/ Modifying Native Code
```
Comment (kbrandwijk) [inline blank/README.md:28]:
```suggestion
## Running/Modifying Native Code
```

File: blank/README.md [hunk 3]
```diff
@@ -19,18 +19,17 @@
...
 
-This project can be run from a web browser or the Expo client app. You may find that you want to add more native code later on. You can do this by ejecting the project and rebuilding it yourself.
+You can generate native iOS and android projects from your Expo config file (**app.json**/ **app.config.js**) by runnning `npx expo prebuild`. These native projects can then be compiled and run via XCode and Android Studio.
```
Comment (kbrandwijk) [inline blank/README.md:30]:
```suggestion
You can generate native iOS and Android projects from your Expo config file (**app.json**/ **app.config.js**) by runnning `npx expo prebuild`. These native projects can then be compiled and run via XCode and Android Studio.
```

File: with-electron/README.md [hunk 4]
```diff
@@ -20,5 +20,6 @@ For more info [check out the latest docs!](https://github.com/expo/expo/blob/mas
 ## 📝 Notes
 
 - [Known issue with app entry](https://github.com/expo/expo-cli/issues/1385)
+- [background on 'require not defined' error](https://github.com/electron/electron-quick-start/issues/463#issuecomment-869219170). If needed, this fix can be applied after running `yarn expo-electron customize` to allow customization of additional electron files.
```
Comment (kbrandwijk) [inline with-electron/README.md:23]:
```suggestion
- [Background on 'require not defined' error](https://github.com/electron/electron-quick-start/issues/463#issuecomment-869219170). If needed, this fix can be applied after running `yarn expo-electron customize` to allow customization of additional electron files.
```

File: with-maps/README.md [hunk 5]
```diff
@@ -14,17 +14,16 @@
...
 
 - Install packages with `yarn` or `npm install`.
-- Run `yarn start` or `npm start` to start the bundler.
+- Run `npx expo start` or `npm start` to start the bundler.
 - Open the project in a React runtime to try it:
   - iOS: [Client iOS](https://itunes.apple.com/app/apple-store/id982107779)
   - Android: [Client Android](https://play.google.com/store/apps/details?id=host.exp.exponent&referrer=blankexample)
 
-## Adding Native Code
+## Running/ Modifying Native Code
```
Comment (kbrandwijk) [inline with-maps/README.md:22]:
```suggestion
## Running/Modifying Native Code
```

File: with-maps/README.md [hunk 6]
```diff
@@ -14,17 +14,16 @@
...
 
-This project can be run from Expo client app. You may find that you want to add more native code later on. You can do this by ejecting the project and rebuilding it yourself.
+You can generate native iOS and android projects from your Expo config file (**app.json**/ **app.config.js**) by runnning `npx expo prebuild`. These native projects can then be compiled and run via XCode and Android Studio.
```
Comment (kbrandwijk) [inline with-maps/README.md:24]:
```suggestion
You can generate native iOS and Android projects from your Expo config file (**app.json**/ **app.config.js**) by runnning `npx expo prebuild`. These native projects can then be compiled and run via XCode and Android Studio.
```

File: with-moti/README.md [hunk 7]
```diff
@@ -33,16 +33,13 @@
 Moti is built with TypeScript and has first-class support. Here's how to add TS support to your project:
 
 - Rename `App.js` ➜ `App.tsx`
-- Run `expo start` -- TypeScript will be automatically configured.
+- Run `npx expo start` -- TypeScript will be automatically configured.
 
 > 💡 Learn more about [TypeScript in Expo](https://docs.expo.dev/guides/typescript/)
 
-### Adding Native Code
+## Running/ Modifying Native Code
```
Comment (kbrandwijk) [inline with-moti/README.md:40]:
```suggestion
## Running/Modifying Native Code
```

File: with-moti/README.md [hunk 8]
```diff
@@ -33,16 +33,13 @@
...
 
-This project can be run from a web browser or the Expo client app. You may find that you want to add more native code later on. You can do this by ejecting the project and rebuilding it yourself.
-
-- Run `expo eject` to create the native projects.
-- You can still run your project in the web browser or Expo client, you just won't be able to access any new native modules you add.
+You can generate native iOS and android projects from your Expo config file (**app.json**/ **app.config.js**) by runnning `npx expo prebuild`. These native projects can then be compiled and run via XCode and Android Studio.
```
Comment (kbrandwijk) [inline with-moti/README.md:42]:
```suggestion
You can generate native iOS and Android projects from your Expo config file (**app.json**/ **app.config.js**) by runnning `npx expo prebuild`. These native projects can then be compiled and run via XCode and Android Studio.
```

File: with-yarn-workspaces/README.md [hunk 9]
```diff
@@ -15,14 +15,15 @@
   </a>
 </p>
 
-Demonstrates the use of `expo-yarn-workspaces` with Expo.
-This example installs a monorepo with two applications, both using two separate custom packages.
+Demonstrates the use of a Yarn Workspace Monorepo with Expo.
+This example installs a monorepo with an application using a separate custom packages.
```
Comment (kbrandwijk) [inline with-yarn-workspaces/README.md:19]:
```suggestion
This example installs a monorepo with an application using a separate custom package.
```

**Review by kbrandwijk — CHANGES_REQUESTED** [review]

**Review by kbrandwijk — APPROVED** [review]

**Pull request merged**